src/RISCV/ArchDesc.hs

```diff
@@ -150,17 +150,19 @@ fromString str = ArchDesc { has_xlen_32     = True
                           }
   where rawSplit = splitOneOf "_zx" (map toLower str)
         archStrings = filter (\x -> not $ null x) rawSplit
-        rv64 = (head archStrings) =~ "rv64"
-        i = (head archStrings =~ "i") || (head archStrings =~ "g")
-        m = (head archStrings =~ "m") || (head archStrings =~ "g")
-        s = (head archStrings =~ "s") || (head archStrings =~ "g")
-        a = (head archStrings =~ "a") || (head archStrings =~ "g")
-        f = (head archStrings =~ "f") || (head archStrings =~ "g")
-        d = (head archStrings =~ "d") || (head archStrings =~ "g")
-        icsr = elem "icsr" archStrings || (head archStrings =~ "g")
-        ihpm = elem "ihpm" archStrings || (head archStrings =~ "g")
-        ifencei = elem "ifencei" archStrings || (head archStrings =~ "g")
-        c = head archStrings =~ "c"
-        n = head archStrings =~ "n"
+        archHead = case archStrings of []  -> ""
+                                       h:t -> h
```

Comment:
Just out of curiosity, could this also be?
`_ -> head archStrings`

Reply:
Or maybe:
`x -> head x`

Reply:
That would reintroduce the unsafe head though, and presumably trigger the same CI warning, unless it's smart enough to notice the empty list has already been matched.

Reply:
Ah ok, makes sense. It was just out of curiosity ;)